ValkeyFullTextSearch.md

```diff
@@ -0,0 +1,392 @@
+│                                                       │ Pipeline           │  │
+│                                                       │                    │  │
+│                                                       │ 1. Split on punct. │  │
+│                                                       │ 2. Case conversion │  │
+│                                                       │ 3. Stop word filter│  │
+│                                                       │ 4. Stemming        │  │
+│                                                       └──────────┬─────────┘  │
+│                                                                  │            │
+│                                                                  ▼            │
+│                                                       ┌────────────────────┐  │
+│                                                       │ Term Storage in    │  │
+│                                                       │ Prefix/Suffix Trees│  │
+│                                                       └────────────────────┘  │
+│                                                                               │
+└───────────────────────────────────────────────────────────────────────────────┘
+
+```
+## Specification
+
+Each ```TEXT``` field has the a set of configurables some control the process to convert a string into a vector of terms, others control the contents of the generated index.
+
+| Metadata          | Type              | Meaning                                                                    | Default Value |
+| ----------------- | ----------------- | -------------------------------------------------------------------------- | ------------- |
+| Punctuation       | vector<character> | Characters that separate words during the tokenization process.            |
+| Stop Words        | vector<String>    | List of stop words to be removed during decomposition                      |
+| Stemming Language | Enumeration       | Snowball stemming library language control                                 |
+| Suffix Tree       | Boolean           | Controls the presence/absence of a suffix tree.                            |
+| Word Offsets      | Boolean           | Indicates if the postings for a word contain word offsets or just a count. |
+
+### Extension of the ```FT.CREATE``` command
+
+Clauses are provided to control the configuration. 
+If supplied before the ```SCHEMA``` keyword then the default value of the clause is changed for any text fields declared in this command.
+If supplied as part of an individual field declaration, i.e., after the ```SCHEMA``` keyword, then it sets the configurable for just that field.
+
+```
+[PUNCTUATION <string>]
+```
+The characters of this string are used to split the input string into words. Note, the splitting process allows escaping of input characters using the usual backslash notation. This string cannot be empty. Default value is: _TBD_
```

Comment:
This is language dependent, so like the problems global language defaults in stopwords (see below), the default punctuation should probably be linked with stemming language